pkgs/leak_tracker/lib/src/leak_tracking/_primitives/_retaining_path/_connection.dart

```diff
@@ -18,16 +18,23 @@ class Connection {
   final VmService service;
 }
 
+/// Connects to vm service protocol.
+///
+/// If it is not found, tries to start it.
 Future<Connection> connect() async {
   _log.info('Connecting to vm service protocol...');
 
-  final info = await Service.getInfo();
+  var info = await Service.getInfo();
+
+  if (info.serverWebSocketUri == null) {
+    info = await Service.controlWebServer(enable: true, silenceOutput: true);
```

Comment:
A _log.info message here 'Starting vm service server' might be useful here .

Reply:
Thanks. Updated this in other PR.